core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -534,11 +549,14 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     // Do not change this code without running the K8s integration suites
     val executorsToDecommission = executorsAndDecomInfo.flatMap { case (executorId, decomInfo) =>
       // Only bother decommissioning executors which are alive.
+      // Keep executor decommission info in case executor started, but not registered yet
       if (isExecutorActive(executorId)) {
         scheduler.executorDecommission(executorId, decomInfo)
         executorsPendingDecommission(executorId) = decomInfo
         Some(executorId)
       } else {
+        unKnownExecutorsPendingDecommission.put(executorId,
```

Comment:
What if the executor is actually killed by the driver right before the decom request comes?

Reply:
Is the concern that the entry in the map will be around for a registration request to come in, that never happens ?
If yes, this can also happen when the executor startup fails - for example.

This should be fine, given the cache is bounded, right ?

Reply:
My concern is actually the race condition between "executor killed by driver" and "executor decommissioned". If "executor killed by driver" happens right before "executor decommissioned", then we'd mistakenly put the executor into `unKnownExecutorsPendingDecommission`. But yes, the cache is bounded as well as this only a race condition, so I think it should be fine.